content/en/blog/_posts/2022-08-29-csi-inline-volumes-ga.md

```diff
@@ -0,0 +1,106 @@
+
+CSI Inline Volumes are similar to other ephemeral volume types, such as `configMap`, `downwardAPI` and `secret`. The important difference is that the storage is provided by a CSI driver, which allows the use of ephemeral storage provided by third-party vendors. The volume is defined as part of the pod spec and follows the lifecycle of the pod, meaning the volume is created once the pod is scheduled and destroyed when the pod is destroyed.
+
+## What's new in 1.25?
+
+There are a couple of new bug fixes related to this feature in 1.25, and the [CSIInlineVolume feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) has been locked to `True` with the graduation to GA. There are no new API changes, so users of this feature during beta should not notice any significant changes aside from these bug fixes.
+
+- [#89290 - CSI inline volumes should support fsGroup](https://github.com/kubernetes/kubernetes/issues/89290)
+- [#79980 - CSI volume reconstruction does not work for ephemeral volumes](https://github.com/kubernetes/kubernetes/issues/79980)
+
+## When to use this feature
+
+CSI inline volumes are meant for simple local volumes that should follow the lifecycle of the pod. They may be useful for providing secrets, configuration data, or other special-purpose storage to the pod from a CSI driver.
+
+A CSI driver is not suitable for inline use when:
+- The volume needs to persist longer than the lifecycle of a pod
+- Volume snapshots, cloning, or volume expansion are required
+- The CSI driver requires `volumeAttributes` that should be restricted to an administrator
+
+## How to use this feature
+
+In order to use this feature, the `CSIDriver` spec must explicitly list `Ephemeral` as one of the supported `volumeLifecycleModes`. Here is a simple example from the [CSI host-path driver](https://github.com/kubernetes-csi/csi-driver-host-path).
+
+```
+apiVersion: storage.k8s.io/v1
+kind: CSIDriver
+metadata:
+  name: hostpath.csi.k8s.io
+spec:
+  volumeLifecycleModes:
+  - Persistent
+  - Ephemeral
+  podInfoOnMount: true
+  fsGroupPolicy: File
+```
+
+Any pod spec may then reference that CSI driver to create an inline volume, as in this example.
+
+```
+kind: Pod
+apiVersion: v1
+metadata:
+  name: my-csi-app-inline
+spec:
+  affinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: topology.hostpath.csi/node
+            operator: Exists
+  containers:
+    - name: my-frontend
+      image: busybox
+      volumeMounts:
+      - mountPath: "/data"
+        name: my-csi-volume
+      command: [ "sleep", "1000000" ]
+  volumes:
+    - name: my-csi-volume
+      csi:
+        driver: hostpath.csi.k8s.io
+```
+
+If the driver supports any volume attributes, these may also be provided as part of the pod spec.
+
+```
+      csi:
+        driver: example.csi.k8s.io
```

Comment:
Is this valid:
```suggestion
        driver: block.csi.vendor.example
```
?

If so, let's illustrate that driver IDs don't have to be official Kubernetes project ones.